### ISSUE TITLE
Add overview to new monorepo to "Developer" section of docs

### ISSUE THREAD

**Issue by jkhartshorne** [pushkin-consortium/pushkin#319]
Something that explains (or points to a tutorial on) changesets, etc.

Basically, if somebody had never seen a monorepo, didn't know about changesets, etc., what would they need to read to make sense of things. 

Asking for a friend. :)

**Comment by jkhartshorne** [comment]
@jessestorbeck -- maybe assign this to someone?

**Comment by jessestorbeck** [comment]
Some of this is covered in the [contributions](https://pushkin-consortium.github.io/pushkin/4.0/developers/contributions/) section, but a "repo orientation" section might be nice too and have some additional information, e.g. Q: "What does `pyproject.toml` do?"; A: "We use the Python package manager Poetry in conjunction with MkDocs to build our documentation; `pyproject.toml` tells Poetry how to set up the build environment for the docs."

@hunterschep @ayang21 -- This seems natural to include in [Getting started on development](https://pushkin-consortium.github.io/pushkin/4.0/developers/getting-started-on-development/). Maybe as the second `#`-level heading?

**Comment by jessestorbeck** [comment]
I've take a stab at this in 67d101579d479df7fc1cbb2793bdf3802cb53420. It's folded into working on the linting setup though, so not currently ready for a PR. It looks like this though:

<img width="1236" alt="Screenshot 2024-04-09 at 6 31 05 PM" src="https://github.com/pushkin-consortium/pushkin/assets/126021561/5a903df3-0d30-4e02-b990-232fdc209c70">

And with the annotation expanded:

<img width="1249" alt="Screenshot 2024-04-09 at 6 31 17 PM" src="https://github.com/pushkin-consortium/pushkin/assets/126021561/2eeffd11-5ca8-4eb5-9dc3-88dab7dfc741">

I went back and forth about what was the best way to format it. I tried a diagram with Mermaid (like the contribution flowchart) and that was terrible. Annotations seemed like the best way not to overwhelm the page with text.
